lightning/src/ln/channelmanager.rs

```diff
@@ -429,19 +429,21 @@ pub struct ChannelDetails {
 }
 
 macro_rules! handle_error {
-	($self: ident, $internal: expr) => {
+	($self: ident, $internal: expr, $their_node_id: expr, $locked_channel_state: expr) => {
```

Comment:
Hmm, this does end up taking an extra lock in the happy case. Maybe split the macro into handle_error_locked and handle_error_unlocked?

Reply:
They were 2 callsites where we may don't have an error, modify a bit to avoid superfluous lock tacking if this is the case. Better than yet another error macro

Reply:
Hmm? No, nearly ever message-handling callsite this is the case?

Reply:
Ah yeah corrected, it's a bit ugly rn but ofc I'm still planning a dry-up following PR